cpp/src/io/utilities/config_utils.cpp

```diff
@@ -80,4 +87,36 @@ bool is_stable_enabled() { return is_all_enabled() or get_env_policy() == usage_
 
 }  // namespace nvcomp_integration
 
-}  // namespace cudf::io::detail
+inline std::mutex& host_mr_lock()
+{
+  static std::mutex map_lock;
+  return map_lock;
+}
+
+inline rmm::host_async_resource_ref default_pinned_mr()
+{
+  static rmm::mr::pinned_host_memory_resource default_mr{};
+  return default_mr;
+}
+
+inline auto& host_mr()
```

Comment:
Required: You need this. See https://github.com/rapidsai/rmm/issues/826 

Probably a good idea to wrap it in a macro. See https://github.com/rapidsai/rmm/blob/branch-24.04/include/rmm/detail/export.hpp 

```suggestion
__attribute__((visibility("default")))
inline auto& host_mr()
```